nexus/reconfigurator/execution/src/dns.rs

```diff
@@ -271,13 +271,43 @@ pub fn blueprint_internal_dns_config(
             ) => (ServiceName::InternalNtp, address.port()),
             BlueprintZoneType::Clickhouse(
                 blueprint_zone_type::Clickhouse { address, .. },
-            ) => (ServiceName::Clickhouse, address.port()),
+            )
+            | BlueprintZoneType::ClickhouseServer(
+                blueprint_zone_type::ClickhouseServer { address, .. },
+            ) => {
+                // Add the HTTP and native TCP interfaces for ClickHouse data
+                // replicas. This adds the zone itself, so we need to continue
+                // back up to the loop over all the Omicron zones, rather than
+                // falling through to call `host_zone_with_one_backend()`.
+                let http_service = if matches!(
+                    &zone.zone_type,
+                    BlueprintZoneType::Clickhouse(_)
+                ) {
+                    ServiceName::Clickhouse
+                } else {
+                    ServiceName::ClickhouseServer
+                };
+
+                // Safety: This only fails if the ServiceName isn't one of the
+                // ClickHouse types, which is guaranteed by this match arm. It
+                // may also fail if we add a zone with the same ID more than
+                // once, but as with the call to `host_zone_with_one_backend()`
+                // below, that should not be possible at this point in the
+                // code, since the IDs of the zones in the blueprint should be
+                // unique.
```

Comment:
I do like this comment better, but I'm still a little nervous about unwrapping. The only non-test callers I found of this function are `deploy_dns` (in this file, already returns a `Result`) and `reconfigurator-cli` (which also already return a `Result`). Could we bubble this error out instead?

Reply:
Fair enough, I didn't actually look at the callers :) Will bubble up shortly.

Reply:
Converted to an internal error in 2ad4db9b4